show/fgnhg.py

```diff
@@ -129,41 +146,94 @@ def hash_view(nhg):
-                            t_dict = state_db.get_all(state_db.STATE_DB, key)
-                    nhg_prefix = "FG_ROUTE_TABLE|" + nhg_prefix
-                else:
-                    nhg_prefix = "FG_ROUTE_TABLE|" + nhg_prefix
-                    t_dict = state_db.get_all(state_db.STATE_DB, nhg_prefix)
+                    if (len(str(bankid)) == 3):
+                        displayed_banks.append(str(bankid))
 
-                vals = sorted(set([val for val in t_dict.values()]))
+                for i in range (0, len(displayed_banks), 8):
+                    bank_output = bank_output + " ".join(displayed_banks[i:i+8]) + "\n" 
 
+                bank_output = bank_output + "\n"
+                table.append([nhg_prefix_report, nhip, bank_output])
+
+        click.echo(tabulate(table, header, tablefmt = "simple"))
+
+    else:
+        header = ["FG NHG Prefix", "Next Hop", "Hash buckets"]
+        try:
+            fg_nhg_member_table = config_db.get_table('FG_NHG_MEMBER')
+        except Exception as e: 
+            ctx.fail("FG_NHG_MEMBER entries not present in config_db")
+            click.echo(e)
+            exit()
+
+        alias_list = []
+        nexthop_alias = {}
+
+        for nexthop, nexthop_metadata in fg_nhg_member_table.items():
+            alias_list.append(nexthop_metadata['FG_NHG'])
+            nexthop_alias[nexthop] = nexthop_metadata['FG_NHG']
+
+        if nhg not in alias_list:
+            ctx.fail ("Please provide a valid NHG alias")
+
+        else:
+            nhip_prefix_map = {}
+            for nhg_prefix in table_keys:
+                bank_dict = {}
+                t_dict = state_db.get_all(state_db.STATE_DB, nhg_prefix)
+                vals = sorted(set([val for val in t_dict.values()]))
                 for nh_ip in vals:
                     bank_ids = sorted([int(k) for k, v in t_dict.items() if v == nh_ip])
+                    nhip_prefix_map[nh_ip.split("@")[0]] = nhg_prefix
                     bank_ids = [str(x) for x in bank_ids]
+
                     if nhg_prefix in output_dict:
                         output_dict[nhg_prefix].append(nh_ip.split("@")[0])
                     else:
                         output_dict[nhg_prefix] = [nh_ip.split("@")[0]]
                     bank_dict[nh_ip.split("@")[0]] = bank_ids
 
-                nhg_prefix_report = (nhg_prefix.split("|")[1])
                 bank_dict = OrderedDict(sorted(bank_dict.items()))
+                output_bank_dict = {}
+                for nexthop, banks  in bank_dict.items():
+                    if nexthop in nexthop_alias:
+                        if nexthop_alias[nexthop] == nhg:
+                            output_bank_dict[nexthop] = banks
+                    else:
+                        ctx.fail ("state_db and config_db have FGNHG prefix config mismatch. Check device config!");
+                        sys.exit(1)
```

Comment:
@kktheballer seems to be unreachable section

Reply:
@kktheballer `sys.exit(1)` is not reachable, since `ctx.fail` will exit context immediately